### PR TITLE
Making cross-compilation easier

### PR DESCRIPTION
**Pull request by jpellegrini**
@egallesio : 

I have been cross-compiling STklos for wireless routers (and it works really well - STklos has a very short startup time, and that makes a huge difference on some devices!)

But some recent changes made cross-compilation a bit more difficult. Looking into it, I see it would actually be nice if a few changes were applied to the `Makefile`s and also to `tmpcomp`/`tmpgenlex`.

### 1 -  Use `$STKLOS_BINARY`, not `src/stklos` during compilation
    
This is a very small change, but is crucial for cross-compilation.
    
When the variable `STKLOS_BINARY` is set, it will be used (during cross-compilation, it will be the locally installed stklos); when it is not set, then `${prefix}/src/stklos` is used.
    
Also, in `tmpcomp`, the prefix was computed as `../`, relative to utils. But in `tmpgenlex` it was hardcoded. This PR also changes `tmpgenlex` to do the same as `tmpcomp`.

### 2 -  Don't use `$@` when calling make recursively
    
This seems to break cross-copilation.
    
The program `tmpcomp` uses the environment variable `STKLOS_BINARY`  to tell which STklos executable it can use. However, when we use this rule:
  
```
../streams/derived.ostk: ../streams/primitive.ostk:
        (cd ../streams && $(MAKE) $@)
```
    
that variable is not passed recursively to the child `make`, and  the process fails during cross-compilation.
    
If we do this:

```
../streams/derived.ostk:
        (cd ../streams && $(MAKE) derived.ostk)

../streams/primitive.ostk:
        (cd ../streams && $(MAKE) primitive.ostk)
```
    
It (magically?) works.

### 3 - A short text explaining how to cross-compile 

Basically, it says one should set the `STKLOS_BINARY` variable.

### CODE REVIEW

**Comment by egallesio** [comment]
Thanks @jpellegrini,

I have merged your PR.

**Comment by egallesio** [comment]
Really weird, the issue with "$@" BTW
